db/migrate/20200903193441_umm_v_1_7_data_migration.rb

```diff
@@ -31,23 +31,34 @@ def down
   ############################# To 1.7 ########################################
   # Merge the name and GroupPath fields with only one slash between them
   def merge_group_path_and_name_for_1_7(draft)
-    return draft unless draft.dig('Characteristics', 'GroupPath')
-    return draft['Name'] = "#{draft.dig('Characteristics', 'GroupPath')}" if draft.dig('Characteristics', 'GroupPath') && !draft['Name']
+    return draft unless draft['Characteristics']
 
-    draft['Name'] = if draft['Name'].start_with?('/') && draft.dig('Characteristics', 'GroupPath').end_with?('/')
-                      "#{draft.dig('Characteristics', 'GroupPath').chop}#{draft['Name']}"
-                    elsif draft['Name'].start_with?('/') || draft.dig('Characteristics', 'GroupPath').end_with?('/')
-                      "#{draft.dig('Characteristics', 'GroupPath')}#{draft['Name']}"
+    characteristics = draft['Characteristics']
+    return draft unless characteristics.is_a?(Hash) && characteristics['GroupPath']
+
+    group_path = characteristics['GroupPath']
+    if group_path && !draft['Name']
+      draft['Name'] = group_path
+      return draft
+    end
+
+    draft['Name'] = if draft['Name'].start_with?('/') && group_path.end_with?('/')
+                      "#{group_path.chop}#{draft['Name']}"
+                    elsif draft['Name'].start_with?('/') || group_path.end_with?('/')
+                      "#{group_path}#{draft['Name']}"
                     else
-                      "#{draft.dig('Characteristics', 'GroupPath')}/#{draft['Name']}"
+                      "#{group_path}/#{draft['Name']}"
```

Comment:
There might be a chance that the group path already includes the base variable name (Most variables generated with UVG would be like this). If that is the case then the new name might have a double base name. Maybe add a check where 
if group_path.end_with?(draft['Name'])
  "#{group_path}"

Reply:
done